### PR TITLE
Feature/add formulas ec2 gen2 8 50 8 53

### LINKED ISSUE #1037: [🔧 Formula Request]: Add formulas from chapter 8.2.3 (8) from gen-2 concrete Eurocode prEN-1992-1-1.
### Formula Description

Add equation 8.50 to 8.53 from new (gen-2) concrete Eurocode (prEN-1992-1-1) to the project.

### Acceptance Criteria

- [ ] Add the formula to the code
- [ ] Add latex expression to the formula
- [ ] Add unittests
- [ ] Add docstrings
- [ ] Update table in docs with classname(s)

### Dependencies

_No response_

### Confirmation

- [x] I have checked that this formula does not already exist in the issue tracker.
- [x] I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by casper-van-der-krift**
## Description

Add new concrete Eurocode formulas 8.50 t/m 8.53 including tests.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **New Features**
  * Added four new structural design formulas (8.50, 8.51, 8.52, 8.53) from prEN 1992-1-1:2023 for calculating additional tensile force due to shear, tensile chord force, compressive chord force, and tensile chord limits. Each formula includes input validation and mathematical notation support.

* **Tests**
  * Added comprehensive test coverage for the four new formulas with validation and output format verification.

### CODE REVIEW

**Comment by GerjanDorgelo** [comment]
I can do this check, once the pytests are all succesful ;)

**Comment by GerjanDorgelo** [comment]
could you attach an issue to this one? to make sure no-one duplicates this work.

**Comment by GerjanDorgelo** [comment]
@coderabbitai review

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    def __init__(self, v_ed: N, theta: DEG) -> None:
+        r"""[$\N_{Vd}$] Additional tensile force due to shear [$N$].
+
+        prEN 1992-1-1:2023 art 8.2.3 (8) - Formula (8.50)
+
+        Parameters
+        ----------
+        v_ed : N
+            [$V_{Ed}$] Design shear force
+        theta : MM
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: degrees

File: blueprints/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    def _evaluate(v_ed: N, theta: DEG) -> N:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_negative(v_ed=v_ed)
+        raise_if_less_or_equal_to_zero(theta=theta)
+        return abs(v_ed) * cot(x=theta)
+
+    def latex(self, n: int = 3) -> LatexFormula:
+        """Returns LatexFormula object for formula 8.50."""
+        _equation: str = r"|V_{Ed}| \cdot \cot\theta"
+        _numeric_equation: str = latex_replace_symbols(
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py:42]: could you add a numeric_eq_with_units too? see prompt examples

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            [$\theta$] Angle of the compression strut with the longitudinal axis of the member
+        """
+        super().__init__()
+        self.v_ed = v_ed
+        self.theta = theta
+
+    @staticmethod
+    def _evaluate(v_ed: N, theta: DEG) -> N:
+        """Evaluates the formula, for more information see the __init__ method."""
+        raise_if_negative(v_ed=v_ed)
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: the absolute maker in the equation indicates that negatieve values should be ok in this formula.

dont forget to change test accordingly

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_51.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                r"z": f"{self.z:.{n}f}",
+                r"N_{Vd}": f"{self.n_vd:.{n}f}",
+                r"N_{Ed}": f"{self.n_ed:.{n}f}",
+            },
+            False,
+        )
+        _numeric_equation_with_units: str = latex_replace_symbols(
+            _equation,
+            {
+                r"M_{Ed}": rf"{self.m_ed:.3f} \ Nmm",
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_51.py]: this one is locked on 3 decimals. can you make it n?

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+                r"z": f"{self.z:.{n}f}",
+                r"N_{Vd}": f"{self.n_vd:.{n}f}",
+                r"N_{Ed}": f"{self.n_ed:.{n}f}",
+            },
+            False,
+        )
+        _numeric_equation_with_units: str = latex_replace_symbols(
+            _equation,
+            {
+                r"M_{Ed}": rf"{self.m_ed:.3f} \ Nmm",
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: 3 decimals

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_53.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                r"M_{Ed,max}": f"{self.m_ed_max:.{n}f}",
+                r"z": f"{self.z:.{n}f}",
+                r"N_{Ed}": f"{self.n_ed:.{n}f}",
+            },
+            False,
+        )
+        _numeric_equation_with_units: str = latex_replace_symbols(
+            _equation,
+            {
+                r"M_{Ed,max}": rf"{self.m_ed_max:.3f} \ Nmm",
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_53.py]: 3 decimals

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+class Form8Dot52CompressiveChordForceDueToShear(Formula):
+    r"""Class representing formula 8.52 for the calculation of the compressive chord force due to shear."""
+
+    label = "8.52"
+    source_document = PR_EN_1992_1_1_2023
+
+    def __init__(
+        self,
+        m_ed: N,
+        z: DIMENSIONLESS,
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: mm

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    ) -> None:
+        r"""[$F_{cd}$] Calculation of the compressive chord force due to shear [$N$].
+
+        prEN 1992-1-1:2023 art 8.2.3 (8) - Formula (8.52)
+
+        Parameters
+        ----------
+        m_ed : N
+            [$M_{Ed}$] Design moment at the section [$Nmm$].
+        z : DIMENSIONLESS
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: mm

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        super().__init__()
+        self.m_ed = m_ed
+        self.z = z
+        self.n_vd = n_vd
+        self.n_ed = n_ed
+
+    @staticmethod
+    def _evaluate(
+        m_ed: N,
+        z: DIMENSIONLESS,
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: mm

File: tests/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_51.py [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+    @pytest.mark.parametrize(
+        ("representation", "expected"),
+        [
+            (
+                "complete",
+                r"F_{td} = \frac{M_{Ed}}{z} + \frac{N_{Vd} + N_{Ed}}{2} = \frac{120000000.000}{300.000} + \frac{"
+                r"2000.000 + 10000.000}{2} = 406000.000 \ N",
+            ),
+            (
```
Comment (GerjanDorgelo) [inline tests/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_51.py:62]: add test with units please

**Review by GerjanDorgelo — CHANGES_REQUESTED** [review]

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py [hunk 11]
```diff
@@ -0,0 +1,56 @@
+"""Formula 8.18 from prEN-1992-1-1:2023: Chapter 8 - Ultimate limit states (ULS)."""
```
Comment (GerjanDorgelo) [inline blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: 8.50

**Review by GerjanDorgelo — CHANGES_REQUESTED** [review]

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py (hunk 1 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py (hunk 2 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py:42]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py (hunk 3 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_51.py (hunk 4 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_51.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py (hunk 5 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_53.py (hunk 6 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_53.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py (hunk 7 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py (hunk 8 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py (hunk 9 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_52.py]: Resolved!

File: tests/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_51.py (hunk 10 above)
Comment (casper-van-der-krift) [reply tests/codes/eurocode/pr_en_1992_1_1_2023/chapter_8_ultimate_limit_states/test_formula_8_51.py:62]: Added the test with units to all tests of this PR. Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py (hunk 11 above)
Comment (casper-van-der-krift) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: Resolved!

File: blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py (hunk 3 above)
Comment (GerjanDorgelo) [reply blueprints/codes/eurocode/pr_en_1992_1_2023/chapter_8_ultimate_limit_states/formula_8_50.py]: can you add a test with negative input? including a complete latex. to show how it (should) work

**Review by GerjanDorgelo — APPROVED** [review]
